apps/sms/views/shared/js/utils.js

```diff
@@ -711,6 +711,57 @@
     },
 
     /**
+     * Returns a rate limited function, useful to reduce the number of function
+     * calls when listening to user input (keypress) or browser events.
+     * @params {function} func - The function that should be rate-limited
+     * @params {number=300} delay - Min amount of time (ms) between two calls
+     * @param {Boolean=false} preventfirstCall - Prevent first synchronous call 
+     * when true.
+     * @param {Boolean=false} preventLastCall - Prevent last trailing call
+     * when true.
+     * @returns {function} - The rate limited function that will call the
+     *  original function
+     */
+    throttle: function(func,
+                       delay = 300, 
+                       preventfirstCall = false, 
```

Comment:
Having one parameter for all is a better way of handling options but it adds one line per default assignment. This is a trade off. It seems that we can't use yet [Destructuring assignement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Function_argument_defaults) here because it is only available in 41.
If using options I would prefer to initialize delay this way (0 must be a valid delay but would be treated as false) :
```javascript
var delay = (typeof options.delay !== undefined) ? options.delay : 300;
```

What do you think ?